gobblin-cluster/src/main/java/org/apache/gobblin/cluster/FsJobConfigurationManager.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.gobblin.util.ClassAliasResolver;
+import org.apache.gobblin.util.ConfigUtils;
+import org.apache.gobblin.util.ExecutorsUtils;
+
+
+/**
+ * A {@link JobConfigurationManager} that reads {@link JobSpec}s from a source path on a
+ * {@link org.apache.hadoop.fs.FileSystem} and posts them to an {@link EventBus}.
+ * The {@link FsJobConfigurationManager} has an underlying {@link FsSpecConsumer} that periodically reads the
+ * {@link JobSpec}s from the filesystem and posts an appropriate JobConfigArrivalEvent with the job configuration to
+ * the EventBus for consumption by the listeners.
+ */
+@Slf4j
+public class FsJobConfigurationManager extends JobConfigurationManager {
+  private static final long DEFAULT_JOB_SPEC_REFRESH_INTERVAL = 60;
+
+  private final long refreshIntervalInSeconds;
+
+  private final ScheduledExecutorService fetchJobSpecExecutor;
+
+  protected final SpecConsumer _specConsumer;
+
+  private final ClassAliasResolver<SpecConsumer> aliasResolver;
+
+  private final Optional<MutableJobCatalog> _jobCatalogOptional;
+
+  public FsJobConfigurationManager(EventBus eventBus, Config config) {
+    this(eventBus, config, null);
+  }
+
+  public FsJobConfigurationManager(EventBus eventBus, Config config, MutableJobCatalog jobCatalog) {
+    super(eventBus, config);
+    this._jobCatalogOptional = jobCatalog != null ? Optional.of(jobCatalog) : Optional.absent();
+    this.refreshIntervalInSeconds = ConfigUtils.getLong(config, GobblinClusterConfigurationKeys.JOB_SPEC_REFRESH_INTERVAL,
+        DEFAULT_JOB_SPEC_REFRESH_INTERVAL);
+
+    this.fetchJobSpecExecutor = Executors.newSingleThreadScheduledExecutor(
+        ExecutorsUtils.newThreadFactory(Optional.of(log), Optional.of("FetchJobSpecExecutor")));
+
+    this.aliasResolver = new ClassAliasResolver<>(SpecConsumer.class);
+    try {
+      String specConsumerClassName = GobblinClusterConfigurationKeys.DEFAULT_SPEC_CONSUMER_CLASS;
+      if (config.hasPath(GobblinClusterConfigurationKeys.SPEC_CONSUMER_CLASS_KEY)) {
+        specConsumerClassName = config.getString(GobblinClusterConfigurationKeys.SPEC_CONSUMER_CLASS_KEY);
+      }
+      log.info("Using SpecConsumer ClassNameclass name/alias " + specConsumerClassName);
+      this._specConsumer = (SpecConsumer) ConstructorUtils
+          .invokeConstructor(Class.forName(this.aliasResolver.resolve(specConsumerClassName)), config);
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException
+        | ClassNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void startUp() throws Exception{
+    super.startUp();
+    // Schedule the job config fetch task
+    this.fetchJobSpecExecutor.scheduleAtFixedRate(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          fetchJobSpecs();
+        } catch (InterruptedException | ExecutionException e) {
+          log.error("Failed to fetch job specs", e);
```

Comment:
Can you try a test with error to make sure  the exception is detected and we don't just lose the scheduled execution silently?